extensions-contrib/tdigestsketch/src/main/java/org/apache/druid/query/aggregation/tdigestsketch/sql/TDigestSketchQuantileSqlAggregator.java

```diff
@@ -0,0 +1,229 @@
+          final boolean inputMatches;
+          final VirtualColumn virtualInput = existing.getVirtualColumns()
+                                                     .stream()
+                                                     .filter(
+                                                         virtualColumn ->
+                                                             virtualColumn.getOutputName()
+                                                                          .equals(theFactory.getFieldName())
+                                                     )
+                                                     .findFirst()
+                                                     .orElse(null);
+
+          if (virtualInput == null) {
+            inputMatches = input.isDirectColumnAccess()
+                           && input.getDirectColumn().equals(theFactory.getFieldName());
+          } else {
+            inputMatches = ((ExpressionVirtualColumn) virtualInput).getExpression()
+                                                                   .equals(input.getExpression());
+          }
+
+          if (inputMatches) {
+            // Found existing one. Use this.
+            return Aggregation.create(
+                ImmutableList.of(),
+                new TDigestSketchToQuantilePostAggregator(
+                    name,
+                    new FieldAccessPostAggregator(
+                        factory.getName(),
+                        factory.getName()
+                    ),
+                    quantile
+                )
+            );
+          }
+        }
+      }
+    }
+
+    // No existing match found. Create a new one.
+    final List<VirtualColumn> virtualColumns = new ArrayList<>();
+
+    if (input.isDirectColumnAccess()) {
+      aggregatorFactory = new TDigestSketchAggregatorFactory(
+          histogramName,
+          input.getDirectColumn(),
+          compression
+      );
+    } else {
+      VirtualColumn virtualColumn = virtualColumnRegistry.getOrCreateVirtualColumnForExpression(
+          plannerContext,
+          input,
+          SqlTypeName.FLOAT
+      );
+      virtualColumns.add(virtualColumn);
+      aggregatorFactory = new TDigestSketchAggregatorFactory(
+          histogramName,
+          virtualColumn.getOutputName(),
+          compression
+      );
+    }
+
+    return Aggregation.create(
+        virtualColumns,
+        ImmutableList.of(aggregatorFactory),
+        new TDigestSketchToQuantilePostAggregator(
+            name,
+            new FieldAccessPostAggregator(
+                histogramName,
+                histogramName
+            ),
+            quantile
+        )
+    );
+  }
+
+  private static class TDigestSketchQuantileSqlAggFunction extends SqlAggFunction
+  {
+    private static final String SIGNATURE1 = "'" + NAME + "(column, quantile)'\n";
+    private static final String SIGNATURE2 = "'" + NAME + "(column, quantile, compression)'\n";
```

Comment:
Can you update the docs to mention that optional compression param is supported for the quantile SQL function as well

Reply:
So I thought about this a bit more and I am not sure what is the right thing to do. Consider a query like this:

`SELECT TDIGEST_QUANTILE(column1, 0.1, 100), TDIGEST_QUANTILE(column1, 0.2, 200), TDIGEST_QUANTILE(column1, 0.2, 300) FROM FOO`

This query does the aggregation to generate a sketch on column1 and then applies the post aggregator to compute quantile out of it (3 times). The problem is that the sketch is generated only once using the compression param 100 (from the first `TDIGEST_QUANTILE(column1, 0.1, 100)` and the compression params from the following calls are ignored. 

Is there a way to validate in Druid SQL that all the calls on TDIGEST_QUANTILE() for a column are using the same compression param? The other alternative is to not support compression param for this method and just use the default value of compression.

Reply:
> The problem is that the sketch is generated only once using the compression param 100 (from the first TDIGEST_QUANTILE(column1, 0.1, 100) and the compression params from the following calls are ignored.

If you add a check that the compression parameter matches before reusing the 
aggregator (https://github.com/apache/incubator-druid/pull/8100#discussion_r309914605), does this still occur?

Reply:
Thanks, that worked.